src/main/scala/org/broadinstitute/dsde/firecloud/core/AgoraPermissionHandler.scala

```diff
@@ -0,0 +1,307 @@
+    val readerAgoraPerm=AgoraPermission(
+      user=Some("reader_user"),
+      roles=Some(readerList))
+    val readerFCPerm=readerAgoraPerm.toFireCloudPermission
+    println("For agora reader "+readerAgoraPerm+" the FC is "+readerFCPerm)
+
+    //Owner all 5
+    var ownerAllFive=List[String]()
+    ownerAllFive=(ownerAllFive:+"Read")
+    ownerAllFive=(ownerAllFive:+"Write")
+    ownerAllFive=(ownerAllFive:+"Create")
+    ownerAllFive=(ownerAllFive:+"Redact")
+    ownerAllFive=(ownerAllFive:+"Manage")
+    val agoraAllFivePerm=AgoraPermission(
+      user=Some("agora_all_5_user"),
+      roles=Some(ownerAllFive))
+    val FCAllFive=agoraAllFivePerm.toFireCloudPermission
+    println("For agora all-5 "+agoraAllFivePerm+" the FC is "+FCAllFive)
+
+    //owner all
+    var ownerAllList=List[String]()
+    ownerAllList=(ownerAllList:+"All")
+    val agoraAllPerm=AgoraPermission(
+      user=Some("all_user"),
+      roles=Some(ownerAllList))
+    val FCAll=agoraAllPerm.toFireCloudPermission
+    println("For Agora ALL permission "+agoraAllPerm+" the FC is "+FCAll)
+
+  return true
+  }
+
+
+  def testTranslations() = {
+    println("\n\n\n*****************************")
+    println("NOW TESTING AGORA->FC TRANSLATION")
+    this.testFromAgoraToFC()
+    println("\n\n\n*****************************")
+    println("NOW TESTING RANDOM AGORA->FC TRANSLATION")
+    this.testFromRandomAgoraToFC()
+    println("\n\n\n*****************************")
+    println("NOW TESTING FC->AGORA TRANSLATION")
+    this.testFromFCToAgora()
+
+  }
+
+  // convenience method to translate a FireCloudPermission object to an AgoraPermission object
+  def toAgoraPermission(fireCloudPermission: FireCloudPermission):AgoraPermission = {
+    fireCloudPermission.user match {
+      case None => {
+        // TODO: should throw an error if user is not specified
+        AgoraPermission(None, Some(List.empty))
+      }
+      case Some(u) => AgoraPermission(Some(u), Some(toAgoraRoles(fireCloudPermission.role)))
+    }
+  }
+
+  // convenience method to translate an AgoraPermission to a FireCloudPermission object
+  def toFireCloudPermission(agoraPermission: AgoraPermission):FireCloudPermission = {
+    agoraPermission.user match {
+      case None => {
+        // TODO: should throw an error if user is not specified
+        FireCloudPermission(None, Some("NO ACCESS"))
+      }
+      case Some(u) => FireCloudPermission(Some(u), Some(toFireCloudRole(agoraPermission.roles)))
+    }
+  }
+
+
+  def hasValidUserAndRole(fireCloudPermission: FireCloudPermission):Boolean = {
+    val vRole=this.hasValidRole(fireCloudPermission)
+    val vUser=this.hasValidUser(fireCloudPermission)
+    return (vRole && vUser)
+  }
+
+
+  def hasValidUser(fireCloudPermission: FireCloudPermission):Boolean = {
+    fireCloudPermission.user match {
+      case Some("") => false
+      case None => false
+      //any non-empty user is okay!
+      case _ => true
+    }
```

Comment:
By convention, most uses of case checks would look for the positive first and handle negative secondarily. I would write this something along the lines of:
```
case x if x.nonEmpty => true
case _ => false
```